server/src/main/java/org/apache/iotdb/db/mpp/plan/planner/distribution/SimpleFragmentParallelPlanner.java

```diff
@@ -114,21 +115,25 @@ private void produceFragmentInstance(PlanFragment fragment) {
     // We need to store all the replica host in case of the scenario that the instance need to be
     // redirected
     // to another host when scheduling
-    if ((analysis.getDataPartitionInfo() == null || analysis.getDataPartitionInfo().isEmpty())
-        && (analysis.getStatement() instanceof QueryStatement
-            && ((QueryStatement) analysis.getStatement()).isAggregationQuery())) {
-      // AggregationQuery && no data region, we need to execute this FI on local
-      fragmentInstance.setExecutorAndHost(
-          new QueryExecutor(
-              new TDataNodeLocation()
-                  .setInternalEndPoint(DataNodeEndPoints.LOCAL_HOST_INTERNAL_ENDPOINT)
-                  .setMPPDataExchangeEndPoint(DataNodeEndPoints.LOCAL_HOST_DATA_BLOCK_ENDPOINT)));
+    if (regionReplicaSet == null || regionReplicaSet.getRegionId() == null) {
+      TDataNodeLocation dataNodeLocation = fragment.getTargetLocation();
+      if (dataNodeLocation != null) {
+        fragmentInstance.setExecutorAndHost(new QueryExecutor(dataNodeLocation));
```

Comment:
add some comments about this if branch, tell readers that currently only show queries will enter here.